doc/rtd/topics/boot.rst

```diff
@@ -157,4 +157,93 @@ finished, the ``cloud-init status`` subcommand can help block external
 scripts until cloud-init is done without having to write your own systemd
 units dependency chains. See :ref:`cli_status` for more info.
 
+First Boot Determination
+************************
+
+cloud-init has to determine whether or not the current boot is a first boot or
+not, so that it applies the appropriate configuration.  On a first boot, it
+should run all "per-instance" configuration, whereas on a subsequent boot it
+should run only "per-boot" configuration.  This section describes how
+cloud-init performs this determination, as well as why it is necessary.
+
+When it runs, cloud-init stores a cache of its internal state for use across
+stages and boots.
+
+If this cache is present, then cloud-init has run on this system before.
+[#not-present]_  There are two cases where this could occur.  Most commonly,
+the instance has been rebooted, and this is a second/subsequent boot.
+Alternatively, the filesystem has been attached to a *new* instance, and this
+is a first boot.  The most obvious case where this happens is when an instance
+is launched from an image captured from a launched instance.
+
+By default, cloud-init attempts to determine which case we're running in by
+checking the instance ID in the cache against the instance ID it determines at
+runtime.  If they do not match, then this is a first boot; otherwise, it's a
+subsequent boot.  Internally, cloud-init refers to this behavior as ``check``.
+
+This behavior is required for images captured from launched instances to
+behave correctly, and so is the default which generic cloud images ship with.
+However, there are cases where it can cause problems. [#problems]_ For these
+cases, cloud-init has support for modifying its behavior to trust the instance
+ID that is present in the system unconditionally.  This means that cloud-init
+will never detect a first boot when the cache is present, and it follows that
+the only way to cause cloud-init to detect a first boot is to manually remove
+cloud-init's cache.  Internally, this behavior is referred to as ``trust``.
```

Comment:
I'd just scrap "Internally....".
This is user-facing documentation.

Reply:
Scrap the whole sentence, or just "Internally, "?

(If the former: I would like to keep the names of the behaviours, because I think it makes it easier to know where to refer back to when you're reading the later parts of the documentation, and "in `trust` mode" is more concise than "has `manual_cache_clean` set to True".)